### PR TITLE
[SDK-1108] Corrected link_identifier in open requests

### PR DESCRIPTION
**Pull request by jdee**
This change finishes the work to correctly send link_identifier to the backend instead of app_link_url. It's currently sent as a string to avoid some difficulty serializing 64-bit integers with Poco. That can probably be solved, but for now this seems to work. This included a new source file (SessionEvent.cpp). While adding the file header, I did a general round of copyright updates in the SDK source, LICENSE file and elsewhere.

Also, working with TestBed in the repo is inconvenient when using the MSI. To iterate, you have to rebuild the MSI and reinstall each time, which is tedious, time-consuming and error-prone. This change introduces a project called TestBed-Local, which uses the BranchIO-Static project in the repo. It looks for Poco and other dependencies in C:\Program Files (x86)\Branch SDK, where they are installed by the MSI. To work with this project, first install the MSI. Then you can build any local changes and deploy to the Start menu as with the main TestBed. The second app has a different Branch key (which required a little refactoring to pass it as a preprocessor definition instead of a code literal) and appears separately in the Start menu, to avoid any conflict with the other version. Note the separate example.html there with a link to a different domain. The two instances of TestBed share as much as possible in the way of code and resources.

After the change to use link_identifier properly:

```
2021-01-26-02:05:15.132994|Debug|8752|SessionEvent.cpp:29[setLinkUrl]|Found raw link_click_id: 811645508874040815
2021-01-26-02:05:15.137007|Debug|12040|APIClientSession.cpp:117[sendRequest]|Sending request to /v1/open
2021-01-26-02:05:15.137007|Verbose|12040|APIClientSession.cpp:118[sendRequest]|Event Payload: {"advertising_ids":{"windows_advertising_id":"my-waid"},"app_version":"1.0","branch_key":"key_live_og1djX2V7XDJ7bWkeGnD5enbxAiq4hcG","device_fingerprint_id":"870020622661585394","identity_id":"882433255478349875","lat_val":0,"link_identifier":"811645508874040815","local_ip":"fdb2:2c26:f4e4:0:2c55:d32f:60f2:4ead","mac_address":"00:1c:42:7e:a2:03","os":"Windows 8","os_version":"6.2","sdk":"native","sdk_version":"1.2.0"}
2021-01-26-02:05:15.184102|Debug|12040|APIClientSession.cpp:103[post]|Request sent. Waiting for response.
2021-01-26-02:05:15.277107|Debug|12040|APIClientSession.cpp:144[processResponse]|[3ea940058d304ed692b9e92d8dbde8d5-2021012602] 200 OK
2021-01-26-02:05:15.278012|Verbose|12040|APIClientSession.cpp:155[processResponse]|Response body: {"data":"{\"$og_title\":\"Branch Metrics Web SDK Example App\",\"$marketing_title\":\"Branch Metrics Web SDK Example App\",\"~creation_source\":1,\"$og_description\":\"A basic example to demonstrate some of the ways that the Web SDK can be used\",\"+click_timestamp\":1611626715,\"$desktop_web_open_delay_ms\":\"3000\",\"$og_image_url\":\"http:\/\/branch.io\/img\/logo_icon_black.png\",\"~feature\":\"marketing\",\"+match_guaranteed\":true,\"$desktop_url\":\"https:\/\/jdee.github.io\/example-win32-local.html\",\"~marketing\":true,\"+clicked_branch_link\":true,\"$one_time_use\":false,\"~id\":\"882426973045331286\",\"$canonical_url\":\"https:\/\/jdee.github.io\/example-win32-local.html\",\"+is_first_session\":false,\"~referring_link\":\"https:\/\/rzc2g.app.link\/KHxB9RHGldb\"}","device_fingerprint_id":"870020622661585394","identity_id":"882433255478349875","link":"https:\/\/rzc2g.app.link?%24identity_id=882433255478349875","session_id":"882433279851275655"}
2021-01-26-02:05:15.280077|Debug|12040|Branch.cpp:97[onSuccess]|Branch::SessionCallback() Success       JSON: {"data":{"$canonical_url":"https:\/\/jdee.github.io\/example-win32-local.html","$desktop_url":"https:\/\/jdee.github.io\/example-win32-local.html","$desktop_web_open_delay_ms":"3000","$marketing_title":"Branch Metrics Web SDK Example App","$og_description":"A basic example to demonstrate some of the ways that the Web SDK can be used","$og_image_url":"http:\/\/branch.io\/img\/logo_icon_black.png","$og_title":"Branch Metrics Web SDK Example App","$one_time_use":false,"+click_timestamp":1611626715,"+clicked_branch_link":true,"+is_first_session":false,"+match_guaranteed":true,"~creation_source":1,"~feature":"marketing","~id":"882426973045331286","~marketing":true,"~referring_link":"https:\/\/rzc2g.app.link\/KHxB9RHGldb"},"device_fingerprint_id":"870020622661585394","identity_id":"882433255478349875","link":"https:\/\/rzc2g.app.link?%24identity_id=882433255478349875","session_id":"882433279851275655"}
2021-01-26-02:05:15.286119|Verbose|12040|Request.cpp:68[send]|POST Success
```

**Updated 1/26:**

For manual testing, deploy either TestBed (or both) as usual (Build > Deploy Solution in Visual Studio). Then open the example.html from the same directory and click the link there. You should get `"+match_guaranteed": true` in the response in the app. Note that I can only get this to work with Firefox right now. Chrome and Edge both immediately redirect to the web before you can click the button in the dialog to redirect to the app.

After adding a unit test and cleaning up some things, my previous comment highlighting the change was outdated. Here is the main part of the change: https://github.com/BranchMetrics/cpp-branch-deep-linking-attribution/pull/78/files#diff-34e4ba7ee30070cd99f03dd6d8e98fe0f497f1be67d1393ec8cfff426bca14ccR15-R48.

@BranchMetrics/core-team

### CODE REVIEW

File: BranchSDK/src/BranchIO/Event/SessionEvent.cpp [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+    // The link_identifier is sent in JSON as a number, equal
+    // to the link_click_id query parameter from the inbound
+    // URI (url argument here).
+    try {
+        URI uri(surl); // throws SyntaxException when invalid
+
+        // vector<pair<string, string>>
+        // ordered, not associative container
+	auto queryParams = uri.getQueryParameters();
```
Comment (jdee) [inline BranchSDK/src/BranchIO/Event/SessionEvent.cpp:26]: Here's the meat of the change. This method uses Poco to get the query parameters from the inbound URI and looks for `link_click_id`.

File: BranchSDK/test/SessionEventTest.cpp [hunk 2]
```diff
@@ -0,0 +1,15 @@
+#include <BranchIO/Defines.h>
+#include <BranchIO/Event/SessionEvent.h>
+#include <gtest/gtest.h>
+
+using namespace testing;
+using namespace std;
+using namespace BranchIO;
+
+TEST(SessionOpenEventTest, TestLinkIdentifier)
```
Comment (jdee) [inline BranchSDK/test/SessionEventTest.cpp:9]: Unit test for link_click_id.

File: BranchSDK/src/BranchIO/Event/SessionEvent.cpp [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+#include <Poco/URI.h>
+
+#include <sstream>
+
+using namespace std;
+using namespace Poco;
+
+namespace BranchIO {
+
+BaseEvent& SessionOpenEvent::setLinkUrl(const String &url) {
```
Comment (jdee) [inline BranchSDK/src/BranchIO/Event/SessionEvent.cpp:15]: The main change was this method. There's also a unit test for it.

File: BranchSDK/test/SessionEventTest.cpp (hunk 2 above)
Comment (aaaronlopez) [reply BranchSDK/test/SessionEventTest.cpp:9]: nice

**Comment by aaaronlopez** [comment]
looks good

File: BranchSDK/test/SessionEventTest.cpp (hunk 2 above)
Comment (aaaronlopez) [reply BranchSDK/test/SessionEventTest.cpp:9]: mentioned on JIRA string will be fine for server

**Review by aaaronlopez — APPROVED** [review]

**Review by aaaronlopez — APPROVED** [review]

**Pull request merged**